Error when trying to install `javascript-unknown-ghcjs-xxx` using the tui interface

# Problem Description

A while ago I succesfully installed the compiler `javascript-unknown-ghcjs-9.6.2` using the instructions in the [user guide](https://www.haskell.org/ghcup/guide/#ghc-js-cross-bindists-experimental). That means that I added an additional release channel and installed an emscripten toolchain.

I now updated ghcup using `ghcup upgrade` and saw that a new compiler `javascript-unknown-ghcjs-9.10.0.20240413` is available in the tui interface. Trying to install this new version using the tui interface fails with the following error:

```
Error: [GHCup-00841] Process "sh" with arguments ["./configure",
                             "--prefix=/home/david/.ghcup/ghc/javascript-unknown-ghcjs-9.10.0.20240413",
                             "--target=javascript-unknown-ghcjs"] failed with exit code 1.
```

Relevant logs:

### .ghcup/logs/ghcup.log

```
Debug: Identified Platform as: Linux Ubuntu, 22.04
Debug: last access was 286.363387133s ago, cache interval is 300s
Debug: Decoding yaml at: /home/david/.ghcup/cache/ghcup-0.0.8.yaml
Debug: last access was 286.15251809s ago, cache interval is 300s
Debug: Decoding yaml at: /home/david/.ghcup/cache/ghcup-cross-0.0.8.yaml
```

### .ghcup/logs/ghc-configure.log

```
...
checking whether CC supports -no-pie... yes
checking whether CC supports flags passed by GHC when compiling via C... yes
checking Setting up CFLAGS, LDFLAGS, IGNORE_LINKER_LD_FLAGS and CPPFLAGS... done
checking Setting up CONF_CC_OPTS_STAGE0, CONF_GCC_LINKER_OPTS_STAGE0, CONF_LD_LINKER_OPTS_STAGE0 and CONF_CPP_OPTS_STAGE0... done
checking Setting up CONF_CC_OPTS_STAGE1, CONF_GCC_LINKER_OPTS_STAGE1, CONF_LD_LINKER_OPTS_STAGE1 and CONF_CPP_OPTS_STAGE1... done
checking Setting up CONF_CC_OPTS_STAGE2, CONF_GCC_LINKER_OPTS_STAGE2, CONF_LD_LINKER_OPTS_STAGE2 and CONF_CPP_OPTS_STAGE2... done
checking whether ld.gold supports response files... yes
checking C++ standard library flavour... libstdc++
checking for linkage against 'stdc++'... success
checking for javascript-unknown-ghcjs-otool... no
checking for otool... no
checking for javascript-unknown-ghcjs-install_name_tool... no
checking for install_name_tool... no
checking emsdk version... configure: error: could not determine emsdk version. Perhaps CC is not emcc?
```

# Diagnosis

Installation via the tui doesn't correctly wrap the invocation of `ghcup install` in an invocation of `emconfigure`. This is why the installation fails with the message `checking emsdk version... configure: error: could not determine emsdk version. Perhaps CC is not emcc?`

# Workaround

Manually call `emconfigure ghcup install ghc --set javascript-unknown-ghcjs-9.10.0.20240413` instead of trying to install via the tui.

# Expected Behaviour

Either don't allow the option to install the javascript targets in the tui (which would be acceptable since it is an experimental opt-in feature) or correctly wrap the invocation of the install script with `emconfigure`.

> I think this means that the section in the GHCup user guide about the javascript backend must be updated

That makes sense. Can you provide a PR?

> This behaviour was implemented in the following PR: https://gitlab.haskell.org/ghc/ghc/-/merge_requests/10918

I'd argue the configure script error is highly misleading and this warrants a GHC bug report.

`checking emsdk version... configure: error: could not determine emsdk version. Perhaps CC is not emcc?` doesn't really tell me that I'm running the wrong emscripten version and which one exactly was expected. This is a bad error message. Can you raise a bug report?

--- a/docs/guide.md
+++ b/docs/guide.md
@@ -559,19 +559,24 @@ The next sections explain how to install each cross bindist.
 
 ### GHC JS cross bindists (experimental)
 
-You need the required emscripten JS toolchain:
+You need the required emscripten JS toolchain. GHC JS cross bindists might require you to install a specific
+version of emscripten. If that is the case, then ghcup will display the required emscripten version in the
+pre install message. You can use the following commands to install the emscripten toolchain on your system,
+substituting the required version for the bindist that you want to install.
+(Cf. [GHC-MR 10918](https://gitlab.haskell.org/ghc/ghc/-/merge_requests/10918))
 
 ```sh
 git clone https://github.com/emscripten-core/emsdk.git
 cd emsdk
-./emsdk install latest
-./emsdk activate latest
+./emsdk install VERSION
+./emsdk activate VERSION
 source ./emsdk_env.sh
 ```
 
-Instructions are also here: [Download and install — Emscripten 3.1.43-git (dev) documentation](https://emscripten.org/docs/getting_started/downloads.html).
+Instructions are also here: [Download and install — Emscripten documentation](https://emscripten.org/docs/getting_started/downloads.html).
 
-To install we need to invoke ghcup like so:
+To install you can either use the tui interface by invoking `emconfigure ghcup tui` or
+you can install directly like so:
 
 ```sh
 emconfigure ghcup install ghc --set javascript-unknown-ghcjs-9.6.2

